DESCRIPTION.md

```diff
@@ -15,6 +15,7 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
   - Bumped pyOpenSSL dependency from >=16.2.0,<24.0.0 to >=16.2.0,<25.0.0.
   - Fixed a memory leak in decimal data conversion.
   - Fixed a bug where `write_pandas` wasn't truncating the target table.
+  - Set `ARROW_DEFAULT_MEMORY_POOL=system` as default memory allocator for `pyarrow.Table.to_pandas`.
```

Comment:
may I ask what's the effect of this parameter?

Reply:
This forces choosing `system` memory allocator which is better than `mimalloc` in terms of memory allocation. Voltron team found some bugs when using mimalloc covered [here](https://www.support.voltrondata.com/servicedesk/customer/portal/1/VESA-156) - for instance they found there is additional memory blooming (a 1 MB requests from pyarrow ends up allocating 64MB memory block) and once conversion is done, the allocator does not return memory back to os. In our tests with `system` allocator, we noted it saved additional 10% memory in sproc environment so this is a good choice as a default allocator.

Reply:
thanks for the explanation. can I understand this as a completely internal implementation detail to improve user experience? from feature wise users won't sense any difference

Reply:
I would set and unset it around the specific to pandas conversion.

Reply:
>  from feature wise users won't sense any difference, or only improvement they can sense.
From feature wise there is no difference, they will only choose a `system` memory allocator if no other default is set  for pyarrow operations.

> also since it's an env var, I'm thinking of whether other python packages (not controlled by us) that customers use will be impacted by this change? is there a different way to set this parameter to scope the impact
`ARROW_DEFAULT_MEMORY_POOL` only affects pyarrow an related functions but you do raise a good point here. In general, pyarrow team encourages every customer to test their workload with the 3 available allocators and choose the best. In sproc environment we have noticed `system` to be better in the scenarios ML has been testing. The default preference in sproc makes sense. Since this is a low impact change, I think we should be okay setting the default value to system here. Once the python program execution is done, this setting is erased so user environment is returned back to a clean environment.